libvcx/src/agency_comm/agency_settings.rs

```diff
@@ -0,0 +1,173 @@
+pub static CONFIG_AGENCY_VERKEY: &str = "agency_verkey";
+pub static CONFIG_REMOTE_TO_SDK_DID: &str = "remote_to_sdk_did";
+pub static CONFIG_REMOTE_TO_SDK_VERKEY: &str = "remote_to_sdk_verkey";
+pub static CONFIG_SDK_TO_REMOTE_DID: &str = "sdk_to_remote_did";
+pub static CONFIG_SDK_TO_REMOTE_VERKEY: &str = "sdk_to_remote_verkey";
+static CONFIG_ENABLE_TEST_MODE: &str = "enable_test_mode";
+
+pub static VALID_AGENCY_CONFIG_KEYS: &[&str] = &[
+    CONFIG_AGENCY_ENDPOINT,
+    CONFIG_AGENCY_DID,
+    CONFIG_AGENCY_VERKEY,
+    CONFIG_REMOTE_TO_SDK_DID,
+    CONFIG_REMOTE_TO_SDK_VERKEY,
+    CONFIG_SDK_TO_REMOTE_DID,
+    CONFIG_SDK_TO_REMOTE_VERKEY,
+    CONFIG_ENABLE_TEST_MODE
+];
+
+
+pub static DEFAULT_DID: &str = "2hoqvcwupRTUNkXn6ArYzs";
+pub static DEFAULT_VERKEY: &str = "FuN98eH2eZybECWkofW6A9BKJxxnTatBCopfUiNxo6ZB";
+pub static DEFAULT_URL: &str = "http://127.0.0.1:8080";
+
+lazy_static! {
+    static ref AGENCY_SETTINGS: RwLock<HashMap<String, String>> = RwLock::new(HashMap::new());
+}
+
+
+fn validate_mandatory_config_val<F, S, E>(val: Option<&String>, err: VcxErrorKind, closure: F) -> VcxResult<u32>
+    where F: Fn(&str) -> Result<S, E> {
+    closure(val.as_ref().ok_or(VcxError::from(err))?)
+        .or(Err(VcxError::from(err)))?;
+
+    Ok(error::SUCCESS.code_num)
+}
+
+fn validate_optional_config_val<F, S, E>(val: Option<&String>, err: VcxErrorKind, closure: F) -> VcxResult<u32>
+    where F: Fn(&str) -> Result<S, E> {
+    if val.is_none() { return Ok(error::SUCCESS.code_num); }
+
+    closure(val.as_ref().ok_or(VcxError::from(VcxErrorKind::InvalidConfiguration))?)
+        .or(Err(VcxError::from(err)))?;
+
+    Ok(error::SUCCESS.code_num)
+}
+
+pub fn set_testing_defaults_agency() -> u32 {
+    trace!("set_testing_defaults_agency >>>");
+
+    // if this fails the test should exit
+    let mut agency_settings = AGENCY_SETTINGS.write().unwrap();
+
+    agency_settings.insert(CONFIG_AGENCY_ENDPOINT.to_string(), DEFAULT_URL.to_string());
+    agency_settings.insert(CONFIG_AGENCY_DID.to_string(), DEFAULT_DID.to_string());
+    agency_settings.insert(CONFIG_AGENCY_VERKEY.to_string(), DEFAULT_VERKEY.to_string());
+    agency_settings.insert(CONFIG_REMOTE_TO_SDK_DID.to_string(), DEFAULT_DID.to_string());
+    agency_settings.insert(CONFIG_REMOTE_TO_SDK_VERKEY.to_string(), DEFAULT_VERKEY.to_string());
+    agency_settings.insert(CONFIG_SDK_TO_REMOTE_DID.to_string(), DEFAULT_DID.to_string());
+    agency_settings.insert(CONFIG_SDK_TO_REMOTE_VERKEY.to_string(), DEFAULT_VERKEY.to_string());
+
+    error::SUCCESS.code_num
+}
+
+pub fn clear_config_agency() {
+    trace!("clear_config_agency >>>");
+    let mut config = AGENCY_SETTINGS.write().unwrap();
+    config.clear();
+}
+
+pub fn validate_agency_config(config: &HashMap<String, String>) -> VcxResult<u32> {
+    trace!("validate_agency_config >>> config: {:?}", config);
+
+    // todo: Since we scope these setting to agency module, these are not really optional anymore!
```

Comment:
They were optional previously so I didn't modify the behaviour. The reason why they were/are optional is because in theory, on `aries-vcx` level you were supposed to have option not to use agency at all. I am not sure if that was implemented only for legacy protocols, or if that still works in the code as we have it today.
I'd suggest to keep this as is to avoid mixing behaviour modifications with refactoring and we can review this behaviour separately. What do you think?

Reply:
Of course, if you think about whether they are optional on `agency_comm` as soon to become standalone module, of course no way. But as per comment above, on global `aries-vcx` level, these were optional.

Reply:
I agree. Making smaller changes is less risky and this refactor is large enough change already.